src/arrayMethodJoin.js

```diff
@@ -5,7 +5,26 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let separatorNew = separator;
+
+    if (separator === undefined) {
+      separatorNew = ',';
+    }
+
+    for (let key of this) {
+      if (key === null || key === undefined) {
+        key = '';
+      }
+      result += key.toString() + separatorNew;
```

Comment:
no need to use .toString(), when you adding something to string its change it to string automatically